### PR TITLE
Separate encrypted and non-encrypted paths

### LINKED ISSUE #127: Separate encrypted lookup path entirely
The double hash multicodec code is not dedicated to encrypted records. Therefore we cannot implicitly assume that it represents one. 

Separate encrypted lookup path by adding `/encrypted` prefix.

Context: 
 * https://github.com/ipfs/specs/pull/421#discussion_r1237036585

### PR DESCRIPTION
**Pull request by gammazero**
Do not rely on dbl-sha2-256 codec to determine if querying for encrypted or non-encrypted multihash. Use URL path instead.

Fixes #127

Requires https://github.com/ipni/dhstore/pull/84

### CODE REVIEW

File: delegated_translator.go [hunk 1]
```diff
@@ -88,17 +89,11 @@ func (dt *delegatedTranslator) find(w http.ResponseWriter, r *http.Request) {
...
-	uri, err := url.ParseRequestURI(findByCid)
-	if err != nil {
-		http.Error(w, "", http.StatusInternalServerError)
-		return
+	var cidUrlParam string
+	if encrypted {
+		cidUrlParam = strings.TrimPrefix(r.URL.Path, "/encrypted/providers/")
+	} else {
+		cidUrlParam = strings.TrimPrefix(r.URL.Path, "/providers/")
 	}
```
Comment (gammazero) [inline delegated_translator.go:97]: Would it be better to just do this?
```suggestion
	cidUrlParam := path.Base(r.URL.Path)
```

**Comment by gammazero** [comment]
Do we still care about separating dhstore backends and non-dhstore backends?  Indexstar and its config could be simplified if not.

File: delegated_translator.go (hunk 1 above)
Comment (masih) [reply delegated_translator.go:97]: I _think_ using `path.Base` would return results for `/encrypted/fish/in/da/sea/<multihash>` right?

**Comment by masih** [comment]
> Do we still care about separating dhstore backends and non-dhstore backends

I don't think we do anymore 👍

**Review by masih — APPROVED** [review]

File: delegated_translator.go (hunk 1 above)
Comment (gammazero) [reply delegated_translator.go:97]: It does what you want: https://go.dev/play/p/wdbYmr6j-l-

Comment (masih) [reply delegated_translator.go]: Oo nice 👍

**Pull request merged**